src/Storage/Repository/PgInstanceRepository.cs

```diff
@@ -506,10 +506,17 @@ private static void AddDateParam(string dateParam, StringValues queryValues, Dic
         {
             foreach (string value in queryValues)
             {
-                string @operator = value.Split(':')[0];
-                string dateValue = value[(@operator.Length + 1)..];
-                string postgresParamName = GetPgParamName($"{dateParam}_{@operator}");
-                postgresParams.Add(postgresParamName, valueAsString ? dateValue : DateTimeHelper.ParseAndConvertToUniversalTime(dateValue));
+                try
```

Comment:
I would suggest we handle this validation at an earlier stage than within the repository class

Reply:
Maybe both? If not, the controller should send the data with more structure, so that the repo class wouldn't need to parse the parameters.

Reply:
I suggest that we don't change existing functionality until the new code has run without problems in production for a couple of weeks.